e2e-tests/src/main.rs

```diff
@@ -1,39 +1,36 @@
 use std::env;
-use std::iter;
 use std::time::Instant;
 
 use clap::Parser;
-use common::create_connection;
-use log::info;
-use sp_core::crypto::Ss58Codec;
-use sp_core::Pair;
-use substrate_api_client::{compose_call, compose_extrinsic, AccountId, XtStatus};
 
 use config::Config;
 
-use crate::utils::*;
-use crate::waiting::{wait_for_finalized_block, wait_for_session};
-
 mod config;
+mod test;
 mod utils;
-mod waiting;
 
 fn main() -> anyhow::Result<()> {
-    if env::var(env_logger::DEFAULT_FILTER_ENV).is_err() {
-        env::set_var(env_logger::DEFAULT_FILTER_ENV, "warn");
-    }
-    env_logger::init();
+    init_env();
 
     let config: Config = Config::parse();
 
-    run(test_finalization, "finalization", config.clone())?;
-    run(test_fee_calculation, "fee calculation", config.clone())?;
-    run(test_token_transfer, "token transfer", config.clone())?;
-    run(test_change_validators, "validators change", config)?;
+    run(test::finalization, "finalization", config.clone())?;
+    run(test::fee_calculation, "fee calculation", config.clone())?;
+    run(test::token_transfer, "token transfer", config.clone())?;
+    run(test::channeling_fee, "channeling fee", config.clone())?;
+    run(test::treasury_access, "treasury access", config.clone())?;
+    run(test::change_validators, "validators change", config)?;
```

Comment:
ince these test run single threaded, ooone thing comes to mind - we could use that fact to implicitely test finalization after changing validators (if there was ever to be a bug there), by moving the test earlier in the suite? For example:

```
    run(test::finalization, "finalization", config.clone())?;
    run(test::fee_calculation, "fee calculation", config.clone())?;
    run(test::change_validators, "validators change", config)?;
    run(test::token_transfer, "token transfer", config.clone())?;
    run(test::channeling_fee, "channeling fee", config.clone())?;
    run(test::treasury_access, "treasury access", config.clone())?;
```

Reply:
reordered